src/nbl/video/COpenGL_Queue.h

```diff
@@ -305,11 +320,10 @@ class COpenGL_Queue final : public IGPUQueue
             {
                 SRequestParams_Fence params;
                 COpenGLFence* glfence = static_cast<COpenGLFence*>(_fence);
-                glfence->setToBeSignaled();
                 params.fence = core::smart_refctd_ptr<COpenGLFence>(glfence);
 
                 auto& req = threadHandler.request(std::move(params));
-                // wait on completion ?
+                threadHandler.template waitForRequestCompletion<SRequestParams_Fence>(req);
```

Comment:
what's this?

Reply:
I realized I have to drop fence  smart ptr after it's not used anymore (after GL cpu wait call was submitted). Just So that queue dispatcher doesn't hold this object forever

Reply:
(see definition of waitForRequestConpletion() in this class)

Reply:
you mean so the `COpenGLSync` is actually made? I dont get this.

Reply:
i dont get what you dont get XD fence smart ptr is written into circular buffer of requests, request is done, then smart ptr is dropped

Reply:
related to 
https://github.com/Devsh-Graphics-Programming/Nabla/pull/28/files#r614412265

Reply:
post `pull/X#discussion_rN` (from discussion page) links, those doesnt work for me

Reply:
https://github.com/Devsh-Graphics-Programming/Nabla/pull/28#discussion_r614412265